File: Day01/Part2.cs
using System;
using System.IO;

var text = await File.ReadAllTextAsync("Day01/input.txt");

var lookup = new Dictionary<string, int>() {
    { "one", 1 },
    { "two", 2 },
    { "three", 3 },
    { "four", 4 },
    { "five", 5},
    { "six", 6 },
    { "seven", 7 },
    { "eight", 8 },
    { "nine", 9 }
};

var answer = text
    .Split("\r\n")
    .Aggregate(0, (acc, state) => {
        var numbers = new List<int>();
        for (var i = 0; i < state.Length; i++) {
            if (int.TryParse($"{state[i]}", out var number)) {
                numbers.Add(number);
            }

            foreach (var (key, value) in lookup) {
                if (state.Substring(i).StartsWith(key)) {
                    numbers.Add(value);
                }
            }
        }

        return acc + numbers.First() * 10 + numbers.Last();
    });

Console.WriteLine(answer);  

File: Day01/Part2.ts
const text = await Deno.readTextFile("./Day01/input.txt")
const numbers:Record<string, number> = {
    one: 1,
    two: 2,
    three: 3,
    four: 4,
    five: 5,
    six: 6,
    seven: 7,
    eight: 8,
    nine: 9
};

const answer = text
    .split("\r\n")
    .reduce((sum, line) => {
        const nums = [];
        for (let i = 0; i < line.length; i++) {
            Object.keys(numbers).forEach(num => {
                if (line.slice(i).startsWith(num)) {
                    nums.push(numbers[num]);
                }
            });
            if (!isNaN(Number(line[i]))) {
                nums.push(parseInt(line[i]));
            }
        }
        return sum + nums[0] * 10 + nums[nums.length - 1];
    }, 0);

console.log(answer);